spec/content/automate/ManageIQ/Infrastructure/VM/Provisioning/Placement.class/__methods__/vmware_best_fit_least_utilized_spec.rb

```diff
@@ -48,11 +48,7 @@
     let(:host3) { FactoryBot.create(:host_vmware, :storages => [ro_storage, storages[2]], :vms => vms[2..4], :ext_management_system => ems) }
     let(:host4) { FactoryBot.create(:host_vmware, :storages => storages[0..2], :vms => vms[2..4], :ext_management_system => ems) }
     let(:host5) { FactoryBot.create(:host_vmware, :storages => [ro_storage, storages[2]], :vms => vms[2..4], :ext_management_system => ems) }
-
-    let(:host_struct) do
-      [MiqHashStruct.new(:id => host1.id, :evm_object_class => host1.class.base_class.name.to_sym),
-       MiqHashStruct.new(:id => host2.id, :evm_object_class => host2.class.base_class.name.to_sym)]
-    end
+    let(:host_struct) { [host1, host2, host4] }
```

Comment:
`host_struct` is changed from array of MiqHashStruct to array of AR objects. Is this by purpose?

Reply:
If you look into the previous spec state, there was a hook, where were the `#eligible_storages` and `#eligible_hosts` methods overridden to return array of specific service models. We changed the automate method to use `eligible_storages` instead of `writable_storages` method and the filtering of the storages is now executed by `eligible_resources` workflow. In this case I had to change the `host_struct`, since it is no more used as the result of hooked`eligible_hosts`, but in the `#find_all_ems_of_type`, which is used inside the `#allowed_hosts` as a part of the eligible workflow. With this change I am able to test the filtering.